packages/sdk/src/storage/useStorage.ts

```diff
@@ -27,27 +27,25 @@ const setItem = async <T>(key: string, value: T | null) => {
   }
 }
 
-const isFunction = <T>(x: T | (() => T)): x is () => T =>
-  typeof x === 'function'
-
 export const useStorage = <T>(key: string, initialValue: T | (() => T)) => {
-  const [data, setData] = useState(() => ({
-    payload: isFunction(initialValue) ? initialValue() : initialValue,
-    state: 'initial',
-  }))
+  const [init] = useState(initialValue)
```

Comment:
I think it doesn't matter in here. But yeah, I think you are right `useRef` is more react-ish in here

Reply:
Oh, no, actually, no. I use the `useState` because initialState can be a function, If I use a `useRef` and the user pass a function to initialValue, `ref.current` will be a function instead of the value. I could do a `if typeof initialValue === 'function'` but useState does it natively for me